parachain/pallets/basic-channel/src/inbound/mod.rs

```diff
@@ -92,7 +92,7 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-		#[pallet::weight(100_000_000)]
+		#[pallet::weight(T::WeightInfo::submit() + message.dispatch_weight)]
```

Comment:
Now that you've got the protocol changes figured out, we need to implement benchmarks for `T::WeightInfo::submit()`.

I think that's going to involve some restructuring of this submit() dispatchable, to factor out the message verification logic into a separate function that can be benchmarked.

After reading https://docs.substrate.io/test/benchmark/, I'd recommend chatting to @claravanstaden who has experience writing benchmarks.